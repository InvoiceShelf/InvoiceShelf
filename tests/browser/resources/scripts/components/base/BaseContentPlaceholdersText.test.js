import { expect, test } from 'vitest'
import { render } from 'vitest-browser-vue'
import BaseContentPlaceholdersText from '@/scripts/components/base/BaseContentPlaceholdersText.vue'

test('BaseContentPlaceholdersText four line animated rounded', async () => {
  const { getByTestId } = render(BaseContentPlaceholdersText, {
    props: {
      animated: true,
      rounded: true,
      'data-testid': 'text-placeholder'
    },
  })
  const textPlaceholder = getByTestId('text-placeholder')
  await expect.element(textPlaceholder).toBeInTheDocument()
  await expect.element(textPlaceholder).toHaveClass('base-content-placeholders-is-animated')
  await expect.element(textPlaceholder).toHaveClass('base-content-placeholders-is-rounded')

  expect(textPlaceholder).toMatchSnapshot()
})
