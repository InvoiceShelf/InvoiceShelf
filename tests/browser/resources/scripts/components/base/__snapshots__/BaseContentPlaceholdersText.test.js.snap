// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseContentPlaceholdersText four line animated rounded 1`] = `
PlaywrightLocator {
  "_container": <body>
    
    
    
  


    <div>
      <div
        class="base-content-placeholders-text base-content-placeholders-is-rounded base-content-placeholders-is-animated"
        data-testid="text-placeholder"
      >
        
        <div
          class="w-full h-full base-content-placeholders-text__line flex flex-col"
        />
        <div
          class="w-full h-full base-content-placeholders-text__line flex flex-col"
        />
        <div
          class="w-full h-full base-content-placeholders-text__line flex flex-col"
        />
        <div
          class="w-full h-full base-content-placeholders-text__line flex flex-col"
        />
        
      </div>
    </div>
  </body>,
  "_parsedSelector": {
    "capture": undefined,
    "parts": [
      {
        "body": [
          {
            "simples": [
              {
                "combinator": "",
                "selector": {
                  "css": "body",
                  "functions": [],
                },
              },
            ],
          },
        ],
        "name": "css",
        "source": "body",
      },
      {
        "body": "[data-testid="text-placeholder"s]",
        "name": "internal:testid",
        "source": "[data-testid="text-placeholder"s]",
      },
    ],
  },
  "_pwSelector": undefined,
  "selector": "body >> internal:testid=[data-testid="text-placeholder"s]",
}
`;
